keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -151,3 +170,171 @@ def get_config(self):
             }
         )
         return config
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation, **kwargs):
+        image = None
+        image = kwargs.get("image")
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomShear()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomShear(bounding_box_format='xyxy')`"
+            )
+        height, width, _ = image.shape
+        image = tf.expand_dims(image, axis=0)
+
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+            dtype=self.compute_dtype,
+        )
+        x, y = transformation
+        # apply horizontal shear
+        if x is not None:
+            bounding_boxes = self._apply_horizontal_transformation_to_bounding_box(
+                bounding_boxes, x
+            )
+        # apply vertical shear
+        if y is not None:
+            bounding_boxes = self._apply_vertical_transformation_to_bounding_box(
+                bounding_boxes, y
+            )
+        # clip bounding boxes value to 0-image height and 0-image width
+        bounding_boxes = RandomShear.clip_bounding_box(height, width, bounding_boxes)
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source="xyxy",
+            target=self.bounding_box_format,
+            images=image,
+            dtype=self.compute_dtype,
+        )
+        return bounding_boxes
+
+    @staticmethod
+    def clip_bounding_box(height, width, bounding_boxes):
```

Comment:
lets give these an _ in their names

Reply:
Didn't understand. you want to add _ in the fn name or arguments?

Reply:
Function name to make them private - sorry that wasn’t clear!